chianti-atomic/ChiantiPy
# Make ChiantiPy an Astropy affiliated package

## Addresses Issue #11

**Not ready to be merged!** 

This pull requests implements the [Astropy affiliated package template](http://docs.astropy.org/en/stable/development/affiliated-packages.html), including adding the [astropy-helpers submodule](https://github.com/astropy/astropy-helpers).
### Brief Summary

Only very minor changes were made to the actual code (see 4). Nearly all changes deal with setup, structure, and docs.
1. Add package template: setup.py now a bit more general, all package specific info now in setup.cfg, main directory name changed from chianti->ChiantiPy to be consistent with package name and avoid naming conflicts in setup, all licenses (i.e. for ChiantiPy and fortranformat) in `licenses/` directory.
2. All routines in `ChiantiPy/` moved to `ChiantiPy/tools` and all imports adjusted appropriately. Now four main modules: `tools`, `core`, `Gui`, and `FortranFormat`. 
3. Lots of docstring formatting to be consistent with [Astropy doc guidelines](http://astropy.readthedocs.io/en/latest/development/docguide.html). 
4. Very few and minor changes to code in `ChiantiPy/tools` (e.g. changing 0/1 to bool True/False where appropriate). 
### Test PR Locally

Note that Astropy is now a dependency.
1. [Checkout the PR](https://help.github.com/articles/checking-out-pull-requests-locally/)
2. Install and build documentation
   
   ```
   python setup.py build
   python setup.py install
   python setup.py build_sphinx
   ```
3. View the docs at `$CHIANTIPY_ROOT/docs/build/html/index.html`
### Possible Roadmap for Astropy Integration

Can be added/changed. Just some ideas!
- [x] Implement basic package template (this PR)
- [ ] Format doc strings for nice looking Sphinx/RTD docs consistent with Astropy/NumPy guidelines (somewhat addressed in this PR)
- [ ] [Astropy units](http://docs.astropy.org/en/stable/units/) for all quantities returned from ChiantiPy/calculations within ChiantiPy
- [ ] Additional features? [Astropy.tables](http://docs.astropy.org/en/stable/table/index.html)?
